src/middlewared/middlewared/plugins/usage.py

```diff
@@ -137,6 +137,16 @@ async def gather_ha_stats(self, context):
             'ha_licensed': await self.middleware.call('failover.licensed'),
         }
 
+    async def gather_directory_service_stats(self, context):
+        config = await self.middleware.call('ldap.config')
+        return {
+            'directory_services': {
+                'state': await self.middleware.call('directoryservices.get_state'),
+                'kerberos_realm_populated': bool(config['kerberos_realm']),
+                'has_samba_schema': config['has_samba_schema'],
```

Comment:
these are specific to ldap so perhaps it should be in its own ldap dict? thoughts @anodos325